### PR TITLE
Support JSON objects when evaluating claims in regex policy

### LINKED ISSUE #11514: Cannot use a nested claim value for Regex comparison in a Regex Policy in Authorization Services
### Describe the bug

If I add a custom mapper in my client to put in a user attribute such that my JWT access token contains a section like this:

```json
{
  "userinfo": {
      "tenant": "abc"
  }
}
```

and then I go and create a regex policy for comparing this value with a Target Claim `userinfo.tenant` and regex pattern `^abc$` ,  then the policy still always evaluates to false.

I'm guessing this is because the regex policy code is trying to find an attribute `{"userinfo.tenant": "abc"}` kind of value in the access token.

I also tried putting the target claim as `userinfo[tenant]`, `userinfo##tenant` and `userinfo\.tenant` but none of them worked.

What WORKED was not nesting the `tenant` key-value pair and changing the regex policy to read a non nested value.

Either this is bug, or I'm missing a piece of documentation that mentions how to read nested claims in the regex policy thing.

Could someone please confirm the issue or suggest a fix ?

### Version

17.0.1

### Expected behavior

The nested claim value can be read and the regex comparision works properly for valid values.

### Actual behavior

Nested claims are not working in regex policy

### How to Reproduce?

Create a regex policy in Authorization services that tries to compare with a nested claim value.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #11514

Documentation: https://github.com/keycloak/keycloak-documentation/pull/1566

### CODE REVIEW

**Comment by abstractj** [comment]
@pedroigor I will be happy to merge after @vmuzikar or @DGuhr approval.

**Review by DGuhr — APPROVED** [review]

**Pull request merged**
